ignite/metrics/nlp/bleu.py

```diff
@@ -146,59 +166,128 @@ def _corpus_bleu(
         for refs, hyp in zip(references, candidates):
             # For each order of ngram, calculate the numerator and
             # denominator for the corpus-level modified precision.
-            for i in range(1, self.ngrams_order + 1):
+            for i in range(1, ngrams_order + 1):
                 numerator, denominator = modified_precision(refs, hyp, i)
                 p_numerators[i] += numerator
                 p_denominators[i] += denominator
 
+            # Calculate the hypothesis lengths
+            hyp_lengths.append(len(hyp))
+
+            # Calculate the closest reference lengths.
+            ref_lengths.append(_closest_ref_length(refs, len(hyp)))
+        return (p_numerators, p_denominators, hyp_lengths, ref_lengths)
+
+    @staticmethod
```

Comment:
Same here, let’s remove the static decorator. Sorry for that, and use self.smoother and self.weigths.